### ISSUE TITLE
Implementing Leonidas on IntelliJ using Psi

### ISSUE THREAD

**Issue by amirsagiv83** [TechnionYP5777/IntelliJ-Spartanizer#23]
Try and implement Leonidas on IntelliJ IDEA, using the built-in library of IntelliJ, The Psi library.
Try and compile it, then using it to build tippers for the future project.

Here we will update the rest of the group for our progress.

**Comment by RoeyMaor** [comment]
We found that PSI has very rich Util classes that provide a lot of functionality, things are looking good.

**Comment by amirsagiv83** [comment]
Me and @RoeyMaor talked with @orimarco, in order to study a little more the structure of Leonidas.
We figured out that we better start with the Matcher.java file.
We decided that this mission can be divided to two different tasks.

1. Investigate how to transform a string into Psi AST - I'm in-charge
2. Translating Matcher.java to PsiElements with the support of the utils we found - Roey is in-charge.

We will continue updating about our progress as soon as possible.

**Comment by amirsagiv83** [comment]
After a little investigation, I came to the the conclusion that using the Psi library separately from a plugin project is not possible, I'll explain,
If we want to create a PsiFile (or any other Psi element) We have to make a connection to a Project.
In my case, in order to create a PsiFile from a text (a string maybe), we will need to create a PsiFileFactory, and than use it to create a PsiFile from text.
I will try to do that hopefully tomorrow, and I'll update on the results.

**Comment by amirsagiv83** [comment]
I managed to get some results!
I wrote a little plugin to try and build an AST with the Psi library from a **STRING**.
So, I wrote a Class that extends AnAction with 2 methods:

![selection_001](https://cloud.githubusercontent.com/assets/15860047/20809567/1daf72de-b80f-11e6-88b3-2ad395911701.png)
![selection_002](https://cloud.githubusercontent.com/assets/15860047/20809573/2352e4b4-b80f-11e6-8426-468d0c611491.png)

I updated the plugin.xml so I'll have a button in the Code section of IntelliJ to execute this code.
I wanted to see in my console the result: a Psi Tree. and this is exactly what I got!
I created a PsiJavaFile element, printed his children, and his main child's children and the result is:
![selection_003](https://cloud.githubusercontent.com/assets/15860047/20809649/6389c4da-b80f-11e6-89f6-a8a3759cdb89.png)
**ignore the last line of the output**

So, to sum up, I found out that it is possible to create a tree from a string. it's a little weird way in my opinion, maybe it won't be intuitive and easy to get used to, but it is definitely possible!

**Comment by amirsagiv83** [comment]
Moving this issue to the next sprint, should be ready with conclusions in a week.
